ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/WebSocketTest.kt

```diff
@@ -51,19 +51,36 @@ class WebSocketTest : ClientLoader() {
     fun testCancel() = clientTests(listOf("Apache", "Android", "Js", "iOS")) {
         config {
             install(WebSockets)
+        }
 
-            test { client ->
-                io.ktor.client.tests.utils.assertFailsWith<CancellationException> {
-                    withTimeout(1000) {
-                        client.webSocket("$TEST_WEBSOCKET_SERVER/websockets/echo") {
-                            repeat(10) {
-                                send(Frame.Text("Hello"))
-                                delay(250)
-                            }
+        test { client ->
+            io.ktor.client.tests.utils.assertFailsWith<CancellationException> {
+                withTimeout(1000) {
+                    client.webSocket("$TEST_WEBSOCKET_SERVER/websockets/echo") {
+                        repeat(10) {
+                            send(Frame.Text("Hello"))
+                            delay(250)
                         }
                     }
                 }
             }
         }
     }
+
+    @Test
+    fun testEchoWSS() = clientTests(listOf("Apache", "Android", "Js", "iOS")) {
+        config {
+            install(WebSockets)
+        }
+
+        test { client ->
+            client.webSocket("wss://echo.websocket.org") {
+                outgoing.send(Frame.Text("PING"))
+                val frame = incoming.receive()
+                if (frame is Frame.Text) {
+                    assertEquals("PING", frame.readText())
+                }
```

Comment:
Is it possible for some reason for frame not to be Frame.Text because of some network problem?
Maybe `assert(frame is Frame.Text && frame.readText() == "PING")`

Reply:
It's not possible and we can perform the strict check here